### ISSUE TITLE
can't write strings to wsgi.errors

### ISSUE THREAD

**Issue by bipface** [pgjones/hypercorn#98]
passing a string to `environ['wsgi.errors'].write` raises an error (`TypeError: a bytes-like object is required`)

according to [the WSGI specification](https://peps.python.org/pep-3333/#input-and-error-streams) (and according to the [older spec](https://peps.python.org/pep-0333/#input-and-error-streams)) the signature should be `write(str)`.

tested with hypercorn 0.14.3 on python 3.10.6

example.py:
```py
def wsgi_app(environ, start_response):
        environ['wsgi.errors'].write('foo')
        start_response('200 OK', [])
        return ()
```

launch and test:
```sh
python3 -m hypercorn example:wsgi_app
curl http://127.0.0.1:8000
```

result:
```
[2023-01-18 19:49:05 +1100] [8477] [ERROR] Error in ASGI Framework
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/hypercorn/asyncio/task_group.py", line 23, in _handle
    await app(scope, receive, send, sync_spawn, call_soon)
  File "/usr/local/lib/python3.10/dist-packages/hypercorn/app_wrappers.py", line 50, in __call__
    await self.handle_http(scope, receive, send, sync_spawn, call_soon)
  File "/usr/local/lib/python3.10/dist-packages/hypercorn/app_wrappers.py", line 82, in handle_http
    await sync_spawn(self.run_app, environ, partial(call_soon, send))
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/dist-packages/hypercorn/app_wrappers.py", line 104, in run_app
    for output in self.app(environ, start_response):
  File "example.py", line 2, in wsgi_app
    environ['wsgi.errors'].write('foo')
TypeError: a bytes-like object is required, not 'str'
```

for reference, running the same example with Gunicorn produces no error

**Comment by sinloss** [comment]
Same issue here.

**Comment by pgjones** [comment]
Fixed in 125bb002903c0e2d60ab6cb36d00dba3cfad6d03
